src/EA.Weee.RequestHandlers.Tests.Unit/Admin/Aatf/GetAatfInfoByAatfIdRequestHandlerTests.cs

```diff
@@ -0,0 +1,102 @@
+        {
+            authorization = AuthorizationBuilder.CreateUserWithAllRights();
+            mapper = new AatfMap(A.Fake<IMap<Domain.UKCompetentAuthority, UKCompetentAuthorityData>>(),
+                A.Fake<IMap<Domain.AatfReturn.AatfStatus, Core.AatfReturn.AatfStatus>>(),
+                A.Fake<IMap<Domain.AatfReturn.AatfSize, Core.AatfReturn.AatfSize>>(),
+                A.Fake<IMap<AatfAddress, AatfAddressData>>());
+            dataAccess = A.Dummy<IGetAatfsDataAccess>();
+            
+            handler = new GetAatfInfoByAatfIdRequestHandler(authorization, mapper, dataAccess);
+        }
+
+        [Fact]
+        public async Task HandleAsync_WhenUserCannotAccessInternalArea_ThrowsSecurityException()
+        {
+            // Arrange
+            IWeeeAuthorization unuthorization = new AuthorizationBuilder().DenyInternalAreaAccess().Build();
+
+            GetAatfInfoByAatfIdRequestHandler handler = new GetAatfInfoByAatfIdRequestHandler(
+                unuthorization,
+                A.Dummy<IMap<Aatf, AatfData>>(),
+                A.Dummy<IGetAatfsDataAccess>());
+
+            // Act
+            Func<Task> testCode = async () => await handler.HandleAsync(A.Dummy<GetAatfById>());
+
+            // Asert
+            await Assert.ThrowsAsync<SecurityException>(testCode);
+        }
+
+        [Fact]
+        public async Task HandleAsync_ProvideAatfId_AatfShouldBeMapped()
+        {
+            // Arrange
+            Guid id = Guid.NewGuid();
+            string name = "fake name";
+            string approvalNumber = "1234";
+
+            Aatf aatf = new Aatf(name, A.Fake<Domain.UKCompetentAuthority>(), approvalNumber, A.Fake<Domain.AatfReturn.AatfStatus>(), A.Fake<Operator>(), A.Fake<AatfAddress>(), A.Fake<Domain.AatfReturn.AatfSize>(), DateTime.Now);
+            A.CallTo(() => dataAccess.GetAatfById(id)).Returns(aatf);
+
+            // Act
+            var result = await handler.HandleAsync(new GetAatfById(A.Dummy<Guid>()));
+
+            // Assert
+            result.ApprovalNumber = approvalNumber;
+            result.Name = name;
+        }
+
+        [Fact]
+        public async Task HandleAsync_GivenGetAatfsReturnRequest_DataAccessFetchIsCalled()
+        {
+            IWeeeAuthorization authorization = new AuthorizationBuilder().AllowInternalAreaAccess().Build();
+
+            await handler.HandleAsync(A.Dummy<GetAatfById>());
+            A.CallTo(() => dataAccess.GetAatfById(A.Dummy<Guid>())).MustHaveHappened(Repeated.Exactly.Once);
+        }
+
+        [Fact]
+        public async Task HandleAsync_ProvideNonExistantAatfId_ReturnsException()
+        {
+            Aatf returnData = null;
+
+            A.CallTo(() => dataAccess.GetAatfById(A.Dummy<Guid>())).Returns(returnData);
+
+            Func<Task> action = async () => await handler.HandleAsync(A.Dummy<GetAatfById>());
+
+            await Assert.ThrowsAsync<ArgumentException>(action);
+        }
+    }
+}
```

Comment:
Add test to check that the value returned from the mapper is the value returned from the HandleAsync method